packages/xstate-wallet/src/ui/approve-budget-and-fund-workflow.tsx

```diff
@@ -60,14 +49,23 @@ export const ApproveBudgetAndFund = (props: Props) => {
     </Flex>
   );
 
-  switch (current.value.toString()) {
-    case 'waitForUserApproval':
-      return waitForUserApproval();
-    case 'createLedgerStartState':
-      return waitForUserApproval({waiting: true});
-    case 'waitForPreFS':
-      return waitForPreFS;
-    default:
-      return <div>todo</div>;
+  if (current.matches('waitForUserApproval')) {
```

Comment:
is it here?

Reply:
I believe so. The relevant feature is [typestates](https://xstate.js.org/docs/guides/typescript.html#typestates), which lets `state.matches` narrow the type of `state.context` -- `state.matches` is typed as [a type guard](https://github.com/davidkpiano/xstate/blob/6b3953ded9b445de77ff39669f28eb05f359b53c/packages/core/src/State.ts#L284-L286).